src/components/AddBook.js

```diff
@@ -1,23 +1,72 @@
-import React from 'react';
+import React, { useState } from 'react';
 import {
   Form, Button, Row, Col,
 } from 'react-bootstrap';
+import { useDispatch } from 'react-redux';
+import { addBook, fetchBooks } from '../redux/books/booksSlice';
 
-const AddBook = () => (
-  <Form className="add-form">
-    <h3>ADD NEW BOOK</h3>
-    <Row className="align-items-center">
-      <Col md={4}>
-        <Form.Control type="text" placeholder="Book Title" />
-      </Col>
-      <Col md={4}>
-        <Form.Control type="text" placeholder="Book Author" />
-      </Col>
-      <Col md={4}>
-        <Button variant="primary" size="lg" block>Add Book</Button>
-      </Col>
-    </Row>
-  </Form>
-);
+const AddBook = () => {
+  const dispatch = useDispatch();
+  const [title, setTitle] = useState('');
+  const [author, setAuthor] = useState('');
+  const [category, setCategory] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const handleAddBook = async () => {
+    if (!title || !author || !category) {
+      setErrorMessage('Please fill in all the fields before adding the book.');
+      return;
+    }
+
+    try {
+      await dispatch(addBook({ title, author, category }));
+      setTitle('');
+      setAuthor('');
+      setCategory('');
+      setErrorMessage('');
+      await dispatch(fetchBooks());
+    } catch (error) {
+      setErrorMessage('Error adding book. Please try again later.');
+    }
+  };
+
+  return (
+    <Form className="add-form">
+      <h3>ADD NEW BOOK</h3>
+      <Row className="align-items-center">
+        <Col md={4}>
+          <Form.Control
+            type="text"
+            placeholder="Book Title"
+            value={title}
+            onChange={(e) => setTitle(e.target.value)}
+          />
+        </Col>
+        <Col md={4}>
+          <Form.Control
+            type="text"
+            placeholder="Book Author"
+            value={author}
+            onChange={(e) => setAuthor(e.target.value)}
+          />
+        </Col>
+        <Col md={4}>
+          <Form.Control
+            type="text"
+            placeholder="Book Category"
+            value={category}
+            onChange={(e) => setCategory(e.target.value)}
+          />
```

Comment:
- [ ] As I already pointed out above about replicating the exact design, your form layout needs a bit of adjustment, the form input and the button are all in the same line from the given design, I'll suggest you update your design to look like this too. :small_airplane: :smile: :wave: 

- - Your design:
![Selection_091](https://github.com/kit0-0/bookstore-react-redux/assets/107632780/8e224c87-3998-4ba1-81d5-ec3620de74b5)

- - Provided design:
![Screenshot from 2023-07-24 15-54-27](https://github.com/kit0-0/bookstore-react-redux/assets/107632780/10ef795a-0d4b-4b91-94c2-b5b3064d18ee)